### PR TITLE
Add `core` target to Makefile

### PR DESCRIPTION
**Pull request by jen20**
This is shorthand to build and install the core of terraform, without touching the providers. Minor wrapper but helpful to people that think in `Makefile`s rather than `go x`!

### CODE REVIEW

**Comment by phinze** [comment]
LGTM

File: Makefile [hunk 1]
```diff
@@ -15,6 +15,12 @@ dev: generate
 quickdev: generate
 	@TF_QUICKDEV=1 TF_DEV=1 sh -c "'$(CURDIR)/scripts/build.sh'"
 
+# Shorthand for quickly building the core of Terraform. Note that some
+# changes will require a rebuild of everything, in which case the dev
+# target should be used.
+core: generate
```
Comment (apparentlymart) [inline Makefile]: Should this be `core-dev` to match with all of the other uses of "dev" to signify "just build for this machine, and don't cross-compile for other targets"?

Comment (radeksimko) [inline Makefile]: :+1: for `core-dev`

Comment (jen20) [inline Makefile]: Yup, I'd agree. I'll make this change.

**Pull request merged**